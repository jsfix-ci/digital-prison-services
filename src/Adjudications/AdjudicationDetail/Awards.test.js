import React from 'react'
import testRenderer from 'react-test-renderer'

import { Awards } from './Awards'

describe('Awards', () => {
  it('should render no award correctly', () => {
    const awards = []

    const wrapper = testRenderer.create(<Awards awards={awards} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should render present award correctly', () => {
    const awards = [
      {
        id: '1',
        sanctionType: 'No association',
        duration: '10 Days',
        effectiveDate: '12/03/2018 - 12:21',
        status: 'Active',
        statusDate: '13/03/2018 - 09:00',
        comment: 'Ongoing',
      },
      {
        id: '2',
        sanctionType: 'Reduction of earnings',
        duration: '1 Month, 10 Days',
        effectiveDate: '15/03/2018 - 10:00',
        status: 'Active',
        statusDate: '16/03/2018 - 08:00',
      },
    ]

    const wrapper = testRenderer.create(<Awards awards={awards} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
