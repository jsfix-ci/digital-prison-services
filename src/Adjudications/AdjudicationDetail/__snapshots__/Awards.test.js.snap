// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Awards should render no award correctly 1`] = `
<div
  className="sc-EHOje jZkQJN"
>
  <h3
    className="sc-htpNat jszPot"
  >
    Awards
  </h3>
  <p>
    No awards available
  </p>
</div>
`;

exports[`Awards should render present award correctly 1`] = `
<div
  className="sc-EHOje jZkQJN"
>
  <h3
    className="sc-htpNat jszPot"
  >
    Awards
  </h3>
  <div
    className="sc-bZQynM eiaTag"
  >
    <div
      className="sc-bdVaJa gkOpjT"
    >
      <div
        className="sc-bwzfXH gDjOjU"
      >
        <strong
          className="label"
        >
          Type
        </strong>
        <p
          className="sc-bxivhb ddqgmM"
        >
          No association
        </p>
      </div>
      <div
        className="sc-bwzfXH gDjOjU"
      >
        <strong
          className="label"
        >
          Length of time
        </strong>
        <p
          className="sc-bxivhb ddqgmM"
        >
          10 Days
        </p>
      </div>
      <div
        className="sc-bwzfXH gDjOjU"
      >
        <strong
          className="label"
        >
          Effective date
        </strong>
        <p
          className="sc-bxivhb ddqgmM"
        >
          12/03/2018 - 12:21
        </p>
      </div>
      <div
        className="sc-bwzfXH gDjOjU"
      >
        <strong
          className="label"
        >
          Status
        </strong>
        <p
          className="sc-bxivhb ddqgmM"
        >
          Active
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa gkOpjT"
    >
      <div
        className="sc-bwzfXH iXLolC"
      >
        <strong
          className="label"
        >
          Comments
        </strong>
        <p
          className="sc-bxivhb ddqgmM"
        >
          Ongoing
        </p>
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM fivOtD"
  >
    <div
      className="sc-bdVaJa gkOpjT"
    >
      <div
        className="sc-bwzfXH gDjOjU"
      >
        <strong
          className="label"
        >
          Type
        </strong>
        <p
          className="sc-bxivhb ddqgmM"
        >
          Reduction of earnings
        </p>
      </div>
      <div
        className="sc-bwzfXH gDjOjU"
      >
        <strong
          className="label"
        >
          Length of time
        </strong>
        <p
          className="sc-bxivhb ddqgmM"
        >
          1 Month, 10 Days
        </p>
      </div>
      <div
        className="sc-bwzfXH gDjOjU"
      >
        <strong
          className="label"
        >
          Effective date
        </strong>
        <p
          className="sc-bxivhb ddqgmM"
        >
          15/03/2018 - 10:00
        </p>
      </div>
      <div
        className="sc-bwzfXH gDjOjU"
      >
        <strong
          className="label"
        >
          Status
        </strong>
        <p
          className="sc-bxivhb ddqgmM"
        >
          Active
        </p>
      </div>
    </div>
  </div>
</div>
`;
